### PR TITLE
WINTERMUTE: Do not optimize out alpha blits on rotation

### PR DESCRIPTION
**Pull request by lolbot-iichan**
This fixes [defect #10684](https://bugs.scummvm.org/ticket/10684): 1-pixel width sprite becomes a big rectangle
if rotation is applied.

Case was originally reproduced with FoxTail game for WME, then reduced
to testcase
https://github.com/tobiatesan/wme_testsuite/tree/master/time_test/packages

### CODE REVIEW

**Comment by lolbot-iichan** [comment]
I wonder if we should use ALPHA_BINARY or ALPHA_FULL here.

File: engines/wintermute/base/gfx/osystem/base_surface_osystem.cpp [hunk 1]
```diff
@@ -433,7 +433,8 @@ bool BaseSurfaceOSystem::drawSprite(int x, int y, Rect32 *rect, Rect32 *newRect,
 	// But no checking is in place for that yet.
 
 	// Optimize by not doing alpha-blits if we lack alpha
-	if (_alphaType == Graphics::ALPHA_OPAQUE && !transform._alphaDisable) {
```
Comment (bluegr) [inline engines/wintermute/base/gfx/osystem/base_surface_osystem.cpp]: Why has the check for transform._alphaDisable been removed?

Comment (lolbot-iichan) [reply engines/wintermute/base/gfx/osystem/base_surface_osystem.cpp]: It was a `if(b!=true){b=true;}` pattern, which has equal effect to just `{b=true;}`, but was harder to read in line of several checks and I was not sure if it is performance-critical here to use `if(b!=true){b=true;}` for a single boolean.

**Comment by bluegr** [comment]
I believe that ALPHA_FULL would be more appropriate in this case, since it does a normal alpha blitting. ALPHA_BINARY is the optimized “transparency or not” blitting.

Since in this case we want to disable optimized blits, ALPHA_FULL should be the right option here

**Comment by lolbot-iichan** [comment]
> Since in this case we want to disable optimized blits, ALPHA_FULL should be the right option here
Ok, fixed!

**Comment by bluegr** [comment]
Thanks! Squashing and merging

**Comment by tobiatesan** [comment]
Thanks. Your work here is first class.

**Pull request merged**